clang/lib/Serialization/ASTReader.cpp

```diff
@@ -10483,6 +10476,13 @@ void ASTReader::finishPendingActions() {
   for (auto *ND : PendingMergedDefinitionsToDeduplicate)
     getContext().deduplicateMergedDefinitonsFor(ND);
   PendingMergedDefinitionsToDeduplicate.clear();
+
+  // For each decl chain that we wanted to complete while deserializing, mark
+  // it as "still needs to be completed".
+  for (unsigned I = 0; I != PendingIncompleteDeclChains.size(); ++I) {
+    markIncompleteDeclChain(PendingIncompleteDeclChains[I]);
+  }
```

Comment:
```suggestion
  for (Decl* D : PendingIncompleteDeclChains)
    markIncompleteDeclChain(D);
```

I know it's sort of preexisting, but while we are here...

Reply:
Yeah, why not